docs/RFCs/011-Sdk-Testing-Platform.md

```diff
@@ -0,0 +1,84 @@
+# RFC 011 - Run dotnet test with Microsoft.Testing.Platform
+
+- [x] Approved in principle
+- [x] Under discussion
+- [ ] Implementation
+- [ ] Shipped
+
+## Current State
+
+Currently, when we run `dotnet test` in CLI, we use vstest as a test runner/driver to run tests in test projects.
+
+## Motivation
+
+With `dotnet test`, users should be able to use [Microsoft testing platform](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-intro?tabs=dotnetcli#microsofttestingplatform-pillars) to run their tests for the sake of improving their experience. They should have the option to opt-in/out this new experience.
+
+The reason for opting-in/out this experience is 
+1. if we were to auto-infer if test projects were using vstest or the testing platform is we will end up with many vulnerabilities, thus making it hard to handle this correctly.
+2. Mixed mode (i.e. having projects using vstest and testing platform in the same solution) will never work as the two platforms have different command line options and different features, thus the mapping will not work as expected.
+
+## Note
+
+We want to design in a way we won't break users and it should be backwards compatible.
```

Comment:
nit: There is no previous solution, so there is nothing to stay backwards compatible with. You probably meant that the feature should be designed in a way that is easy to evolve when new requirements arise, without breaking backwards compatibility.

In other words feature that is future proof, and easy to keep backwards compatible.